lib/openshift/flakes/route_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'base_helper'
+
+module BushSlicer
+
+  # this class should help with parsing route specifications
+  class RouteSpec
+    include Common::Helper
+
+    attr_reader :struct
+    private :struct
+
+    def initialize(struct)
+      @struct = struct
+    end
+
+    module ExportMethods
+      def  host
+        return struct['host']
+      end
+
+      def targetPort
+        return struct.dig('port', 'targetPort')
+      end
+
+      def subdomain
+        return struct['subdomain']
+      end
+
+      def tls_insecureEdgeTerminationPolicy
+        return struct.dig('tls', 'insecureEdgeTerminationPolicy')
+      end
+
+      def tls_termination
+        return struct.dig('tls', 'termination')
+      end
+
+      def to_kind
+        return struct.dig('to', 'kind')
+      end
+
+      def to_name
+        return struct.dig('to', 'name')
+      end
+
+      def to_weight
+        return struct.dig('to', 'weight')
+      end
+
+      def wildcardPolicy
+        return struct['wildcardPolicy']
+      end
```

Comment:
We need `path` as well to parse `spec.path`

Reply:
It's also be better If ca info could be added:  tls_certificate, tls_key, tls_caCertificate, tls_destinationCACertificate

Reply:
@yapei @yanpzhan do you have an example output I can look at because from the route that I've seen, I don't seen those elements.

Reply:
@pruan-rht Here is a example route yaml you can refer to: http://pastebin.test.redhat.com/797253